src/ray/core_worker/transport/sequential_actor_submit_queue.cc

```diff
@@ -91,6 +91,14 @@ SequentialActorSubmitQueue::PopAllOutOfOrderCompletedTasks() {
   return result;
 }
 
+void SequentialActorSubmitQueue::OnClientConnected() {
+  auto head = requests.begin();
+  if (head != requests.end()) {
+    RAY_CHECK(head->first >= next_task_reply_position);
```

Comment:
Discussed offline: we might be able to completely remove `next_task_reply_position` and `out_of_order_completed_tasks`